skimage/measure/_colocalization.py

```diff
@@ -0,0 +1,371 @@
+    p-value : float
+        Two-tailed p-value.
+
+    Notes
+    -------
+
+    Pearson's Correlation Coefficient (PCC) measures the linear correlation
+    between the pixel intensities of the two images. Its value ranges from -1
+    for perfect linear anti-correlation and +1 for perfect linear correlation.
+    The calculation of the p-value assumes that the intensities of pixels in
+    each input image are normally distributed.
+
+    Scipy's implementation of Pearson's correlation coefficient is used. Please
+    refer to it for further information and caveats [1]_.
+
+    .. math::
+        r = \frac{\sum (A_i - m_A_i) (B_i - m_B_i)}
+                 {\sqrt{\sum (A_i - m_A_i)^2 \sum (B_i - m_B_i)^2}}
+
+    where
+        :math:`A_i` is the value of the :math:`i^{th}` pixel in `imgA`
+        :math:`B_i` is the value of the :math:`i^{th}` pixel in `imgB`,
+        :math:`m_A_i` is the mean of the pixel values in `imgA`
+        :math:`m_B_i` is the mean of the pixel values in `imgB`
+
+    A low PCC value does not necessarily mean that there is no correlation
+    between the two channel intensities, just that there is no linear
+    correlation. You may wish to plot the pixel intensities of each of thw two
+    channels in a 2D scatterplot and use Spearman's rank correlation if a
+    non-linear correlation is visually identified [2]_. Also consider if you
+    are interested in correlation or co-occurence, in which case a method
+    involving segmentation masks (e.g. MCC or intersection coefficient) may be
+    more suitable [3]_ [4]_.
+
+    Providing the ROI of only relevant sections of the image (e.g. cells, or
+    particular cellular compartments) and removing noise is important as the
+    PCC is sensitive to these measures [3]_ [4]_.
+
+    References
+    -------
+    .. [1] https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.pearsonr.html
+    .. [2] https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.spearmanr.html
+    .. [3] Dunn, K. W., Kamocka, M. M., & McDonald, J. H. (2011). A practical
+           guide to evaluating colocalization in biological microscopy.
+           American journal of physiology. Cell physiology, 300(4), C723–C742.
+           https://doi.org/10.1152/ajpcell.00462.2010
+    .. [4] Bolte, S. and Cordelières, F.P. (2006), A guided tour into
+           subcellular colocalization analysis in light microscopy. Journal of
+           Microscopy, 224: 213-232.
+           https://doi.org/10.1111/j.1365-2818.2006.01706.x
+    """
+    if roi is None:
+        roi = np.ones_like(imgA)
+    check_numpy_arr(imgA, 'imgA', bool_expected=False)
+    check_numpy_arr(imgB, 'imgB', bool_expected=False)
+    check_numpy_arr(roi, 'roi', bool_expected=True)
+    check_shape_equality_all(imgA, imgB, roi)
+
+    imgA_masked = imgA[roi.astype(bool)]
+    imgB_masked = imgB[roi.astype(bool)]
+    return pearsonr(imgA_masked, imgB_masked)
+
+
+def mcc(imgA, imgB_mask, roi=None):
```

Comment:
```suggestion
def mcc(image, mask, roi=None):
```

In cases with just one image and one mask, please just use `image` and `mask` as the parameter names.